### PR TITLE
rustdoc-json: Clean up serialization and printing.

### PR DESCRIPTION
**Pull request by aDotInTheVoid**
Somewhat a followup to #128963, but makes sense regardless.

- Renames `out_path` to `out_dir` because it's not the path to the JSON, but the directory
  - Also adds a comment explaining `None`
- Renames `write` to `serialize_and_write` because it does both.
  - Also renames the self-profile activity name to be clear this measures both IO cost and serialization CPU cost
  - Expands the timer to cover flushing
- Renames `output` to `output_crate`, to emphasize it's the contents, not the `--output` flag.

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustdoc/json/mod.rs [hunk 1]
```diff
@@ -101,18 +103,21 @@ impl<'tcx> JsonRenderer<'tcx> {
             .unwrap_or_default()
     }
 
-    fn write<T: Write>(
+    fn serialize_and_write<T: Write>(
         &self,
-        output: types::Crate,
+        output_document: types::Crate,
```
Comment (GuillaumeGomez) [inline src/librustdoc/json/mod.rs:108]: Shouldn't it be `output_crate`?

Comment (aDotInTheVoid) [reply src/librustdoc/json/mod.rs]: Oops, yes.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! r=me once CI pass.

**Pull request merged**